src/core/Form/InputClearButton/InputClearButton.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { baseStyles } from './InputClearButton.baseStyles';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import { HtmlButton, HtmlButtonProps } from '../../../reset';
+import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
+import { Icon } from '../../Icon/Icon';
+
+const baseClassName = 'fi-input-clear-button';
+
+const InputClearButtonClassNames = {
+  icon: `${baseClassName}_icon`,
+};
```

Comment:
I get that the object makes the syntax match that of other implementations, but is it necessary here? No harm really in having it, just thought to bring it up.

Reply:
Not explicitly necessary, but having it here makes the implementation also more future proof. Now adding more classNames is easy and naming is easier too. And it's aligned with other components. My personal opinion is to have no exceptions to patterns unless there is a real payoff.

Reply:
Yeah, that makes sense to me too. Let's leave it as it is.

Reply:
Yeah, that makes sense to me too. Let's leave it as it is.

Reply:
Yeah, that makes sense to me too. Let's leave it as it is.